packages/docprovider/src/yprovider.ts

```diff
@@ -95,6 +92,44 @@ export class WebSocketProviderWithLocks
     this._isInitialized = false;
     this._onConnectionStatus = this._onConnectionStatus.bind(this);
     this.on('status', this._onConnectionStatus);
+
+    const state = options.state;
+    if (name != '' || color != '#' || state == null) {
+      if (name == '') {
+        name = getAnonymousUserName();
+      }
+      if (color == '#') {
+        color = '#' + getRandomColor().slice(1);
+      }
+      // only set if this was not already set by another plugin
+      if (currState && currState.name == null) {
```

Comment:
Hi @jess-x, there is an issue here. `currState` doesn't have an attribute `name`. It has an attribute `user` which contains `name` and `color`. This is a small bug that we solve in [this PR](https://github.com/jupyterlab/jupyterlab/pull/10686).